### ISSUE TITLE
site's ssl is rip

### ISSUE THREAD

**Issue by GusCaplan** [google/codeworld#357]

**Comment by cdsmith** [comment]
I'm not sure what you mean by this.  Can you give more details on the problem?

**Comment by GusCaplan** [comment]
@cdsmith your ask certs are expired or like sslv3 or something. I can't tell for sure because of the cert details render issue in chrome beta

**Comment by GusCaplan** [comment]
SSL* not ask

**Comment by stefan-j** [comment]
On Chrome, it shows:

> Issued On	Saturday, December 26, 2015 at 2:00:00 AM
> Expires On	Thursday, December 27, 2018 at 1:59:59 AM

**Comment by GusCaplan** [comment]
![](https://i.imgur.com/qrIDqsW.png)

**Comment by GusCaplan** [comment]
i would hit view cert but the render glitch kicks in hard

**Comment by cdsmith** [comment]
Hmm... Chrome, Firefox and Microsoft Edge are all telling me there are no problems with the certificate.  Which Chrome version are you using?  Perhaps I can install that version and reproduce.

**Comment by xrisk** [comment]
Here's an image from Chrome 53.0.2785.116 (this is the stable channel).

![image](https://cloud.githubusercontent.com/assets/10669859/18711705/d114aaa2-8028-11e6-8bd8-9c6c80f89445.png)

**Comment by cdsmith** [comment]
Okay, so the claim is that your computer doesn't trust the issue.  The issuer is Network Solutions, which should be trusted by default.  I don't suppose you've done something to change your set of trusted root CAs...

Any chance you're using a Mac?  I see a number of reports that MacOS can have problems when there are expired certificates in its local CA store.  For example, https://kb.wisc.edu/page.php?id=45559 gives instructions to fix it.  (They point to a specific cert used by their site, but anything that's expired in that list could be the culprit.)

**Comment by xrisk** [comment]
Yeah, I am in fact running OS X and haven't modified with the Keystore in any way. I am running the brand new OS X release though (from yesterday), so there could be something wrong because of that.

[Chrome cert info for Network Solutions](https://cloud.githubusercontent.com/assets/10669859/18719508/3f3d2d3a-8045-11e6-9e6c-3fa52f158df7.png)

There's only one match for Network solutions in the key store, which is this one.

[System keystore cert for Network Solutions](https://cloud.githubusercontent.com/assets/10669859/18719580/90386ee8-8045-11e6-8c8c-471e903d6e5d.png)

**Comment by cdsmith** [comment]
I'm not really sure how to make progress on this.

I taught at a school yesterday with about 30 MacBooks, and did not see this.  So it's certainly not true on all MacOS systems.  Unfortunately, from looking around to investigate this bug, it seems like errors about SSL certificate chains are a common problem with MacOS, and affect various sites equally.  In all cases I've seen, the answer turns out to be fixing something on the client.

**Comment by cdsmith** [comment]
@schrepfler @GusCaplan @xrisk 

Can one of you check if you're still seeing problems?  I've reinstalled the SSL certificates for the site, and there was a change in one of the certificates when I downloaded them again.

**Comment by xrisk** [comment]
Great, it works now!

**Comment by schrepfler** [comment]
Cert is green now. Cheers.

**Comment by schrepfler** [comment]
From a security point of view however it might be smart to investigate how did the invalid cert end up there and was it put there by mistake and by an authorised individual.

**Comment by cdsmith** [comment]
I wasn't clear.  It wasn't an invalid certificate, but rather one that was missing.  I am pretty sure that what happened was that Apple at some point changed their trusted root certificates, which broke SSL resolution after some updates of Apple operating systems.  Network Solutions then included a new intermediate certificate to re-establish that trust.  That explains everything about this: that it was only broken on MacOS, and even then on some systems, but not others (and in particular, not by the older systems at a school where I teach); and that it was fixed by adding an intermediate cert that wasn't in the chain before.
